view/ViewableData.php

```diff
@@ -358,44 +358,37 @@ public function renderWith($template, $customFields = null) {
 	 */
 	public function obj($fieldName, $arguments = null, $forceReturnedObject = true, $cache = false, $cacheName = null) {
 		if(!$cacheName) $cacheName = $arguments ? $fieldName . implode(',', $arguments) : $fieldName;
-		
+
 		if(!isset($this->objCache[$cacheName])) {
 			// HACK: Don't call the deprecated FormField::Name() method
 			$methodIsAllowed = true;
 			if($this instanceof FormField && $fieldName == 'Name') $methodIsAllowed = false;
-			
+
 			if($methodIsAllowed && $this->hasMethod($fieldName)) {
 				$value = $arguments ? call_user_func_array(array($this, $fieldName), $arguments) : $this->$fieldName();
 			} else {
 				$value = $this->$fieldName;
 			}
-			
-			if(!is_object($value) && ($this->castingClass($fieldName) || $forceReturnedObject)) {
-				if(!$castConstructor = $this->castingHelper($fieldName)) {
+
+			if(!is_object($value) && ($this->castingClass($fieldName))) {
```

Comment:
Now castingClass and castingHelper always return a value, so it's no longer necessary to check for this.